esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1814,17 +1814,28 @@ bool CFileSprayEx::onGetDFUExceptions(IEspContext &context, IEspGetDFUExceptions
     return true;
 }
 
-void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char* srcIP, const char* srcPath,
+void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char* srcIP, const char* srcPath, const char* srcPlane,
     StringBuffer& sourceIPReq, StringBuffer& sourcePathReq)
 {
     StringBuffer sourcePath(srcPath);
-    sourceIPReq.set(srcIP);
-    sourceIPReq.trim();
     sourcePath.trim();
+
     if(srcxml.length() == 0)
     {
+        if (!isEmptyString(srcPlane))
+        {
+            Owned<IPropertyTree> plane =  getStoragePlane(srcPlane);
```

Comment:
I'm slightly concerned about security implications.  This needs to be restricted to drop zones - not any storage plane, otherwise it would allow arbitrary files to be "sprayed" from an existing storage plane.